crypto/keys/pem_signer_factory.go

```diff
@@ -21,17 +21,32 @@ import (
 	"crypto/rsa"
 	"crypto/x509"
 	"fmt"
+	"sync"
 
 	"github.com/golang/protobuf/ptypes"
 	"github.com/golang/protobuf/ptypes/any"
 	"github.com/google/trillian"
 	"github.com/google/trillian/crypto/keyspb"
 )
 
+var (
+	pkcs11Module string
+	pMu          sync.Mutex
+)
+
+// SetPKCS11Module should be called before creating a PEMSignerFactory
+// if the tree is expected to contain a PKCS#11 key
+func SetPKCS11Module(modulePath string) {
+	pMu.Lock()
+	defer pMu.Unlock()
+	pkcs11Module = modulePath
+}
+
 // PEMSignerFactory handles PEM-encoded private keys.
```

Comment:
Would you like to do this in this PR or split it out into another? (I'm fine with either, it seems like there could be some bike-shedding about what it should be called, if the interface is still needed if there is only one implementation etc etc)